source/adios2/toolkit/format/bp1/BP1Writer.cpp

```diff
@@ -56,7 +56,8 @@ void BP1Writer::WriteProcessGroupIndex(
     WriteNameRecord(name, dataBuffer, dataPosition);
 
     // processID in metadata,
-    const uint32_t processID = static_cast<uint32_t>(m_BP1Aggregator.m_RankMPI);
+    const uint32_t processID =
```

Comment:
Why change the cast?  The base type is `int` so the cast to `uint32_t` ensures it's the right size and constness is in the assignment via the lvalue type, so it already has const correctness.